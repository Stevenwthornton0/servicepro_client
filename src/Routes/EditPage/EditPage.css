.serviceEdit {
    width: 80%;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 50px;
}

.serviceEdit > div {
    margin-bottom: 20px;
    display: flex;
    flex-direction: column;
}

.serviceEdit > div > input, .serviceEdit > div > textarea {
    margin-top: 4px;
    font-size: 16px;
    padding: 8px;
    resize: none;
    border-radius: 4px;
    border: 1px solid #424242;
}

.type {
    font-size: 20px;
    font-weight: 600;
    margin-bottom: 0;
}

.editButton {
    width: 100%;
    margin-top: 10px;
    margin-left: 0;
    border: 1px solid lightblue;
    border-radius: 4px;
    font-size: 18px;
    font-weight: 400;
    padding-top: 10px;
    padding-bottom: 10px;
    background-color: lightblue;
}

.editButton:focus {
    outline: none;
}

@media screen and (min-width: 500px) {
    .serviceEdit {
        width: 30%;
        margin-bottom: 40px;
    }

    .editButton:hover {
        background-color: #424242;
        border: 1px solid #424242;
        color: lightblue;
        cursor: pointer;
    }
}