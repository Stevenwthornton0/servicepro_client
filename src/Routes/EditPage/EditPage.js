import React, { Component } from 'react';
import ServicesContext from '../../contexts/ServicesContext';
import ServicesApiService from '../../services/services-api-service';
import './EditPage.css';

class EditPage extends Component {
    static contextType = ServicesContext;

    state = { error: null }

    onUpdateSuccess = () => {
        const { history } = this.props;
        history.push('/services')
    } 

    handleSubmit = ev => {
        ev.preventDefault();
        const { serviceId } = this.props.match.params;
        const { name, about, phone, email } = ev.target;

        this.setState({ error: null })
        ServicesApiService.updateService(
            serviceId, {
            name: name.value,
            about: about.value,
            phone: phone.value,
            email: email.value
        })
            .then(this.onUpdateSuccess())
    }

    componentDidMount() {
        const { serviceId } = this.props.match.params
        ServicesApiService.getService(serviceId)
            .then(service => {
                this.context.setService(service)
                this.context.setServiceUserId(service.user.id)
            })
    }

    componentWillUnmount() {
        this.context.clearService();
    }

    render() {
        const { error, service } = this.context;
        return (
            <form className='serviceEdit' onSubmit={this.handleSubmit}>

                <div role='alert'>
                    {error && <p>{error}</p>}
                </div>

                <div>
                    <p className='type'>Service Type: {service.service_type}</p>
                </div>

                <div>
                    <label>Business Name</label>
                    <input 
                        name='name'
                        type='text'
                        defaultValue={service.name}
                        required
                        id='ProviderRegistration_name'
                    />
                </div>

                <div>
                    <label>Business Description</label>
                    <textarea 
                        required
                        aria-label='Business Description'
                        id='about'
                        cols='30'
                        rows='7'
                        defaultValue={service.about}
                    />
                </div>

                <div>
                    <label>Phone</label>
                    <input 
                        name='phone'
                        type='tel'
                        required
                        id='ProviderRegistration_phone'
                        defaultValue={service.phone}
                    />
                </div>

                <div>
                    <label>Email</label>
                    <input 
                        name='email'
                        type='email'
                        required
                        id='ProviderRegistration_email'
                        defaultValue={service.email}
                    />
                </div>

                <button className='editButton' type='submit'>
                    update
                </button>
            </form>
        )
    }
}

export default EditPage