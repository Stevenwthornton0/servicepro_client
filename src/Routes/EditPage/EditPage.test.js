import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import EditPage from './EditPage';

Enzyme.configure({ adapter: new Adapter() })

describe('EditPage component', () => {
    it('renders without crashing', () => {
        const props = {
                path: "/services/service/:serviceId/edit",
                url: "/services/service/1/edit",
                isExact: true,
                params: {
                    serviceId: "1"
                }
            }
        
        const wrapper = mount(<EditPage match={props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})