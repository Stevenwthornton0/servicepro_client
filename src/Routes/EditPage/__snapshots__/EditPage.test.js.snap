// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPage component renders without crashing 1`] = `
<EditPage
  match={
    Object {
      "isExact": true,
      "params": Object {
        "serviceId": "1",
      },
      "path": "/services/service/:serviceId/edit",
      "url": "/services/service/1/edit",
    }
  }
>
  <form
    className="serviceEdit"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div>
      <p
        className="type"
      >
        Service Type: 
      </p>
    </div>
    <div>
      <label>
        Business Name
      </label>
      <input
        id="ProviderRegistration_name"
        name="name"
        required={true}
        type="text"
      />
    </div>
    <div>
      <label>
        Business Description
      </label>
      <textarea
        aria-label="Business Description"
        cols="30"
        id="about"
        required={true}
        rows="7"
      />
    </div>
    <div>
      <label>
        Phone
      </label>
      <input
        id="ProviderRegistration_phone"
        name="phone"
        required={true}
        type="tel"
      />
    </div>
    <div>
      <label>
        Email
      </label>
      <input
        id="ProviderRegistration_email"
        name="email"
        required={true}
        type="email"
      />
    </div>
    <button
      className="editButton"
      type="submit"
    >
      update
    </button>
  </form>
</EditPage>
`;
